vllm/distributed/device_communicators/shm_broadcast.py

```diff
@@ -154,19 +153,21 @@ def acquire_write(self):
                 written_flag = metadata_buffer[0]
                 if written_flag and read_count != self.buffer.n_reader:
                     # this block is written and not read by all readers
-                    # try to write to the next block
-                    self.current_idx = (self.current_idx +
-                                        1) % self.buffer.max_chunks
-                    if self.current_idx == start_index:
-                        # no empty block found
-                        if time.time(
-                        ) - start_time > VLLM_RINGBUFFER_WARNING_INTERVAL * n_warning:  # noqa
-                            logger.warning(
-                                "No available block found in %s second. ",
-                                VLLM_RINGBUFFER_WARNING_INTERVAL)
-                            n_warning += 1
-                        # wait for a while (0.1 us)
-                        time.sleep(1e-7)
+                    # for writers, `self.current_idx` is the next block to write
+                    # if this block is not ready to write,
+                    # we need to wait until it is read by all readers
+
+                    # wait for a while (0.1 us)
+                    time.sleep(1e-7)
+
+                    # if we wait for a long time, we should warn the user
+                    if time.time(
```

Comment:
we should be using `time.monotonic` everywhere for tracking time intervals.

Reply:
fixed in [004950d](https://github.com/vllm-project/vllm/pull/5801/commits/004950d25d932760f653b0875814ec66a00c636f)